lib/CFRPageActions.jsm

```diff
@@ -92,6 +92,10 @@ class PageAction {
       // After one second, expand
       this._expand(DELAY_BEFORE_EXPAND_MS);
 
+      // Collapse the recommendation on url bar focus in order to free up more
+      // space to display and edit the url
+      this.urlbar.onfocus = this._collapse;
```

Comment:
I believe event listeners are usually added with the method instead of setter to allow for multiple listeners on the same event. This also means we probably want to remove the listener depending on the context.. where here in `show`, we can end up accidentally adding multiple listeners?

From a quick glance, there could potentially be multiple `PageAction` instances per window and each instance could potentially be `show`n multiple times?